src/render/python/integrator_v.cpp

```diff
@@ -77,6 +77,37 @@ MI_VARIANT class PySamplingIntegrator : public SamplingIntegrator<Float, Spectru
             return SamplingIntegrator::render(scene, sensor, seed, spp, develop, evaluate);
     }
 
+    TensorXf render_forward(Scene* scene,
+                            void* params,
+                            Sensor *sensor,
+                            uint32_t seed = 0,
+                            uint32_t spp = 0) override {
+
+        py::gil_scoped_acquire gil;
+        py::function render_forward_override = py::get_override(this, "render_forward");
+
+        if (render_forward_override)
+            return render_forward_override(scene, params, sensor, seed, spp).template cast<TensorXf>();
+        else
+            return SamplingIntegrator::render_forward(scene, params, sensor, seed, spp);
+    }
```

Comment:
I'm unclear how this implementation would be different from just using the `PYBIND11_OVERRIDE` macro (this also applies to `render` and `render_backward`.

Reply:
Yep, I agree and have changed it